### PR TITLE
Null safety migration

### PR DESCRIPTION
**Pull request by justacid**
Hi there, thanks for the package!

This PR adds support for null-safety (see #28).
I have tested the example app on Android and iOS 
and everything seems to be working fine.

Cheers

### CODE REVIEW

**Comment by vanlooverenkoen** [comment]
Very nice! I tought I migrated all my packages yesterday. but indeed the db viewer for moor is something I forgot. I will do the review right away

File: lib/src/screen/moor_table_content_list_screen.dart [hunk 1]
```diff
@@ -152,6 +158,9 @@ class _MoorTableContentListScreenState extends State<MoorTableContentListScreen>
       content: Text(message),
       behavior: SnackBarBehavior.floating,
     );
-    _key.currentState.showSnackBar(snackBar);
+
+    if (_key.currentContext != null) {
```
Comment (vanlooverenkoen) [inline lib/src/screen/moor_table_content_list_screen.dart]: move this to a seperate local variable. So you don't have to do the null check twice

File: lib/src/screen/moor_table_filter_screen.dart [hunk 2]
```diff
@@ -134,17 +134,19 @@ class _MoorTableFilterScreenState extends State<MoorTableFilterScreen>
...
             onPressed: () => Navigator.of(context).pop(),
           ),
         ],
       ),
     );
-    if (result != null)
-      Provider.of<MoorTableFilterViewModel>(_scaffoldKey.currentContext,
-              listen: false)
-          .onWhereColumnSelected(result);
+    if (result != null && _scaffoldKey.currentContext != null) {
```
Comment (vanlooverenkoen) [inline lib/src/screen/moor_table_filter_screen.dart]: move context to a seperate variable

File: lib/src/screen/moor_table_filter_screen.dart [hunk 3]
```diff
@@ -168,9 +170,10 @@ class _MoorTableFilterScreenState extends State<MoorTableFilterScreen>
   Future<void> showEdit(String selectQuery) async {
     final result =
         await DbViewerNavigator.of(context).goToTableFilterEditSql(selectQuery);
-    if (result != null)
-      Provider.of<MoorTableFilterViewModel>(_scaffoldKey.currentContext,
+    if (result != null && _scaffoldKey.currentContext != null) {
```
Comment (vanlooverenkoen) [inline lib/src/screen/moor_table_filter_screen.dart]: move context to a seperate variable

File: lib/src/widget/filter/where/where_string_widget.dart [hunk 4]
```diff
@@ -58,9 +58,11 @@ class WhereStringWidget extends StatelessWidget {
         .onUpdatedWhereClause();
   }
 
-  void _onChangedType(BuildContext context, StringWhereType value) {
-    whereClause.onSelectedType(value);
-    Provider.of<MoorTableFilterViewModel>(context, listen: false)
-        .onUpdatedWhereClause();
+  void _onChangedType(BuildContext context, StringWhereType? value) {
+    if (value != null) {
```
Comment (vanlooverenkoen) [inline lib/src/widget/filter/where/where_string_widget.dart]: if (value == null) return

File: lib/src/widget/filter/where/where_int_widget.dart [hunk 5]
```diff
@@ -59,9 +59,11 @@ class WhereIntWidget extends StatelessWidget {
         .onUpdatedWhereClause();
   }
 
-  void _onChangedType(BuildContext context, IntWhereType value) {
-    whereClause.onSelectedType(value);
-    Provider.of<MoorTableFilterViewModel>(context, listen: false)
-        .onUpdatedWhereClause();
+  void _onChangedType(BuildContext context, IntWhereType? value) {
+    if (value != null) {
```
Comment (vanlooverenkoen) [inline lib/src/widget/filter/where/where_int_widget.dart]: if (value == null) return

File: lib/src/widget/filter/where/where_double_widget.dart [hunk 6]
```diff
@@ -59,9 +59,11 @@ class WhereDoubleWidget extends StatelessWidget {
         .onUpdatedWhereClause();
   }
 
-  void _onChangedType(BuildContext context, DoubleWhereType value) {
-    whereClause.onSelectedType(value);
-    Provider.of<MoorTableFilterViewModel>(context, listen: false)
-        .onUpdatedWhereClause();
+  void _onChangedType(BuildContext context, DoubleWhereType? value) {
+    if (value != null) {
```
Comment (vanlooverenkoen) [inline lib/src/widget/filter/where/where_double_widget.dart]: if (value == null) return

**Review by vanlooverenkoen — CHANGES_REQUESTED** [review]
Thanks for your contribution. I will probably have to test it myself. I have some time this weekend to do the testing as well.

**Comment by justacid** [comment]
Thanks for taking the time! I have addressed your review comments. :)

**Comment by vanlooverenkoen** [comment]
Alright looks good. I will do the extra testing this weekend. Thanks again for your contribution

**Review by vanlooverenkoen — APPROVED** [review]

**Comment by vanlooverenkoen** [comment]
released in 3.0.0

**Pull request merged**